### ISSUE TITLE
width and height with .json

### ISSUE THREAD

**Issue by SwissCheese5** [bbc/audiowaveform#114]
It seems the .json output is the same with or without a width and height, is this intentional?

**Comment by chrisn** [comment]
Yes, the `--width` and `--height` options only apply when creating PNG images. For JSON output you'll need to set the `--zoom` option (or `--pixels-per-second`).

**Comment by SwissCheese5** [comment]
@chrisn I suppose the only way to do this is to average chunks of the data together to get 1 pair of channels to equal x amount of pixels for a specific width? That's disappointing. I'll close this issue and do that i suppose.

**Comment by chrisn** [comment]
The calculation you need to obtain the `--zoom` value for a given image width and audio duration is [here](https://github.com/bbc/audiowaveform/blob/master/src/WaveformGenerator.cpp#L85). Rounding errors mean the resulting JSON isn't exactly the length you requested, but it'll be close.

**Comment by SwissCheese5** [comment]
Thank you.

**Comment by SwissCheese5** [comment]
@chrisn I was successful in using what you linked, however I actually do need it to be pixel perfect. What can I do?

**Comment by chrisn** [comment]
At the moment, only solution is to either truncate or pad the data to the desired length. You're not the first person to request this, see also #41, #104, #108.

**Comment by SwissCheese5** [comment]
very odd. it should be the exact same formula as the png command but without the drawing part.. Hm..
